cmake/Utils.cmake

```diff
@@ -104,7 +104,8 @@ function(xgboost_set_cuda_flags target)
     target_compile_definitions(${target} PRIVATE -DXGBOOST_USE_NVTX=1)
   endif()
 
-  target_link_libraries(${target} PRIVATE CCCL::CCCL)
+  target_link_libraries(${target}
+    PRIVATE CCCL::CCCL CUDA::cudart_static)
```

Comment:
This would maintain the same linking behavior as before

```suggestion
    PRIVATE CCCL::CCCL PUBLIC CUDA::cudart_static)
```

Reply:
Since CCCL is header only. There shouldn't be any linking

Though please let me know if I'm missing something

Reply:
If `CCCL::CCCL` is an interface dep, the downstream user that uses `xgboost::xgboost` from `find_package(XGBoost)` will be asked to link with the CMake target `CCCL::CCCL`. Whether CCCL is a header-only lib is not relevant here.

Reply:
@trivialfis Does XGBoost use any Thrust or CUB structures in the public API ? My guess is that we don't.

Reply:
No, XGBoost defines a C API. Anything written in C++ is not part of it.

We have a document about the C++ headers in the `include` directory but have explicitly emphasized that this is not a stable API. In addition, the only dependency in those headers is dmlc-core.

Reply:
Thanks, then we can keep CCCL and CUDA runtime as `PRIVATE` deps.

Reply:
Ok sounds good. Thanks for the thorough investigations on both of these topics! 🙏

Let's just make sure we are doing the same thing in conda-forge that is being done here. This will help us stay aligned and reduce conda-forge patches over time

Somewhat tangential question, but is Python using the C or C++ interface? Guessing C (through `ctypes`) though wanted to check

Reply:
> but is Python using the C or C++ interface

C